share/completions/npm.fish

```diff
@@ -69,9 +69,13 @@ end
 # use npm completion for most of the things,
 # except options completion (because it sucks at it)
 # and run-script completion (reading package.json is faster).
+# and uninstall (reading dependencies is faster and more robust and yarn needs the functions anyways)
 # see: https://github.com/npm/npm/issues/9524
 # and: https://github.com/fish-shell/fish-shell/pull/2366
-complete -f -c npm -n 'not __fish_npm_needs_option; and not __fish_npm_using_command run; and not __fish_npm_using_command run-script' -a "(__fish_complete_npm)"
+complete -f -c npm -n 'not __fish_npm_needs_option; and not __fish_npm_using_command run; and not __fish_npm_using_command run-script;
+and not __fish_npm_using_command audit;
+and not __fish_npm_using_command r; and not __fish_npm_using_command remove; and not __fish_npm_using_command rm; and not __fish_npm_using_command un;
+and not __fish_npm_using_command uninstall; and not __fish_npm_using_command unlink' -a "(__fish_complete_npm)"
```

Comment:
How about we just fix `__fish_npm_using_command` to check multiple commands?

Just change this

https://github.com/fish-shell/fish-shell/blob/67d1d80f94107e26585d192d194295180cb9f73b/share/completions/npm.fish#L24

to

```fish
if contains -- $cmd[2] $argv
```

Then this can be simplified to

```fish
not __fish_npm_needs_option; and not __fish_npm_using_command run run-script audit r remove rm un uninstall unlink
```

Reply:
will do, the problem does not get that much better tho since i actually forgot to include all the aliases / misspellings for the command (there are only 2/4 for `npm run` and 3/6 for `npm uninstall`).

looking through the code and testing i actually think that this code runs on any tab without doing anything almost ever. i looked and found that i had missed 1 alias that it was auto completing and i don't see literally anything else. the only thing other than commands that are auto completed through this function are superseded and overwritten [here](https://github.com/fish-shell/fish-shell/blob/84545861d030adc46a73e094d7ffcbbc584d1099/share/completions/npm.fish#L121-L126), [here](https://github.com/fish-shell/fish-shell/blob/84545861d030adc46a73e094d7ffcbbc584d1099/share/completions/npm.fish#L114-L116) and [here](https://github.com/fish-shell/fish-shell/blob/4d089e14c4ff941fa74a6f168a4b0241d210de5c/share/completions/npm.fish#L116-L117). removing it will at this point probably only improve overall performance.

Reply:
another issue with this implentation is that it actually doesn't properly work.

here's the generated completion for `npm help`
![image](https://github.com/fish-shell/fish-shell/assets/63159454/79c47766-4b99-4859-80f2-ce90267c614f)

but if i type anything and try to complete that it doesn't want to
![image](https://github.com/fish-shell/fish-shell/assets/63159454/3d2abde7-7075-492a-b046-7b5a5ed69745)